src/link_manager/LinkManagerStateMachineActiveActive.cpp

```diff
@@ -831,6 +849,36 @@ void ActiveActiveStateMachine::LinkProberUnknownMuxWaitLinkUpTransitionFunction(
     switchMuxState(nextState, mux_state::MuxState::Label::Standby);
 }
 
+//
+// ---> LinkProberUnknownMuxWaitLinkDownTransitionFunction(CompositeState &nextState)
+//
+// transition function when entering {LinkProberUnnown, MuxWait, LinkDown} state
+//
+void ActiveActiveStateMachine::LinkProberUnknownMuxWaitLinkDownTransitionFunction(CompositeState &nextState)
+{
+    MUXLOGINFO(mMuxPortConfig.getPortName());
+    if (ps(mCompositeState) != link_prober::LinkProberState::Unknown) {
```

Comment:
Isn't link down the sufficient condition for state to become standby? (same below).

Reply:
Yes, it's sufficient, when there is a link down event we simply switch to standby. 

But here is the edge case, when link down event comes before state machine is activated (before heartbeat is event sent) no action will be taken. We won't get a 2rd link down notification,  so have to rely on link prober events  to trigger the toggles.

Reply:
* In `LinkProberUnknownMuxWaitLinkDownTransitionFunction`, I wanted to remove the mux state check in `handleStateChange()` for link down event. So, we don't have to enter TransitionFunction at all for up -> down event. 

* In `LinkProberUnknownMuxUnknownLinkDownTransitionFunction`, agree with you. That's why in UT I also check if enter by `(unknown, standby, down)` -> `(unknown, unknown, down)`, state setter invoke count won't increase.